ArduSub/Sub.h

```diff
@@ -253,7 +253,8 @@ class Sub : public AP_Vehicle {
     // Baro sensor instance index of the external water pressure sensor
     uint8_t depth_sensor_idx;
 
-    AP_Motors6DOF motors;
+    AP_Motors6DOF *motors;
```

Comment:
```suggestion
    AP_MotorsMatrix *motors;
```
`AP_Motors6DOF` inherits from `AP_MotorsMatrix` as does the existing `AP_MotorsMatrix_6DoF_Scripting`. So by changing the pointer to there common parent it allows us to load either.  

It would be nice to move up another level too to `AP_MotorsMulticopter` in line with copter, but that is not required.

To enable this you will have to add any functions that sub needs from `AP_Motors6DOF` to `AP_MotorsMatrix` as virtual functions.

Reply:
Done.